_drafts/wsl2-as-a-firmware-development-environment.md

```diff
@@ -0,0 +1,485 @@
+  by being compliant here.  It went OK, except that it never prompted me for the
+  name of my home (user) folder and then it named it incorrectly: “jphut” instead of
+  “jp”.  Now, many people would have left it there.  And perhaps I should have. 
+  But “jphut” is _wrong_.  Home is at `/home/jp`, or `c:/users/jp`, or `/users/jp` -
+  this is how it has _always been_.  “No big deal”, I thought. “I’ll fix it!” 
+  
+  It took me 3 hours.  RegEdit.  Reviewing scripts left on Windows forums. 
+  Seriously breaking my install.  Adding symlinks for OneDrive.  It’s been working
+  fine for a few months now, but for all I know I’m out of spec in a weird way that
+  will cause problems.  So, to quote Linus Torvalds, “F@&# you, \[Microsoft\]!”
+- 📎Unsettling paperclip mascot!
+
+### VMWare Workstation Pro
+
+⭐⭐⭐
+
+Pros:
+- With a fully featured VM, you get most of the pros of Linux with access to 
+  Window’s strong points when you need them.
+- USB forwarding, including BLE, works well because you can run a full Linux
+  distro of your choice with all the kernel modules that you need.
+- SSH to a headless VM allows for a snappy, low-latency, experience in your
+  terminal emulator and editor of choice.
+
+Cons:
+- Setup and maintenance.  VMs are complicated, and setting up and maintaining
+  the VM takes time and research.  A common example is the allocation of disk 
+  space.  Generally, you will need to allocate the space for your guest OS up
+  front.  It can be hard to predict exactly how much space you really need, so
+  you may end up needing to expand the virtual disk at some point, requiring
+  steps to be taken from both the host and guest.
+
+## Conclusion
+
+Ultimately, choosing a development environment depends on what helps you to
+be most productive.  Decisions about security, performance,
+features, and stability must be weighed distinctly within each organization and
+project.  Personally, I’ve found a nice balance by using WSL2 on Windows 11. 
+I get all of the security, stability, compatibility, and support of Windows while
+being able to use my preferred tools in a Linux environment.  My primary concern,
+performance, is eased by quantifying that a WSL2 Zephyr build system is only ~12% slower
+than bare metal Linux.  If you compile big Zephyr projects with 40s build times - 
+_pristine instead of incremental_ - 20 times a day (I don’t), WSL2 costs you an extra 1m 30s
+of compile time.  For the tools that my work requires, that 1m 30s is an OK cost
+to pay for all of the benefits that the Windows host provides.
+
+If you’re interested in trying out WSL2 for firmware development, checkout the
+[complete tutorial](https://github.com/intercreate/docs/blob/main/Windows/wsl2.md),
+or stay tuned to the Interrupt where the tutorial will be published TBD.
+
+<!-- TODO: replace the link to WSL2 tutorial when published [complete tutorial]({{ site.baseurl }}{% link _posts/wsl2-firmware-development-setup.md %}) -->
```

Comment:
Thanks for adding this in advance! Jekyll is failing to process this HTML comment, so we'll need to use comment style used by jekyll:

```suggestion
{% comment %}
TODO: replace the link to WSL2 tutorial when published [complete tutorial]({{ site.baseurl }}{% link _posts/wsl2-firmware-development-setup.md %})
{% endcomment %}
```

Usually we have builds on PRs but currently they don't work for PRs from forks due to cloudflare's limitations. On our to-do list, thanks for bearing with us!